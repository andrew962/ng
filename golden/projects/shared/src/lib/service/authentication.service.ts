import { Injectable } from '@angular/core';
import { BehaviorSubject, Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class AuthenticationService {
  isLogged$: Observable<boolean>;
  isLoggedSubject: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
  constructor() {
    this.isLogged$ = this.isLoggedSubject.asObservable();
  }
}
