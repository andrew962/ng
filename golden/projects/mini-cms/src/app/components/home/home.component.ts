import { Component, OnInit } from '@angular/core';
import { AuthenticationService } from 'projects/shared/src/lib/service/authentication.service';

@Component({
  selector: 'app-home',
  templateUrl: './home.component.html',
  styles: [
  ]
})
export class HomeComponent implements OnInit {

  constructor(
    private authenticationService: AuthenticationService
  ) { }

  ngOnInit(): void {
  }

}
