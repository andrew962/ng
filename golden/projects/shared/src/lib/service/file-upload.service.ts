import { Injectable } from '@angular/core';
import { AngularFireStorage } from '@angular/fire/storage';
import { Observable, Subscription } from 'rxjs';
import { concatMap, finalize, tap } from 'rxjs/operators';
import { CurrentUser } from '../interfaces/currentUser.model';
import { AuthenticationService } from './authentication.service';
import { UserService } from './user.service';

@Injectable({
  providedIn: 'root'
})
export class FileUploadService {
  private uid: string;
  user: CurrentUser;
  authSubscription: Subscription;
  downloadURL: Observable<string>;
  constructor(
    private fStorage: AngularFireStorage,
    private auth: AuthenticationService,
    private userService: UserService
  ) {
    this.authSubscription = this.auth.currentUser$.subscribe(u => {
      this.user = u;
      this.uid = u?.Uid
    });
  }
  upload(file, fileName?: string, path: string = 'images') {
    let filePath: string = `${this.uid}/${path}/${fileName}`
    let fileRef = this.fStorage.ref(filePath);
    return this.fStorage.upload(filePath, file).snapshotChanges().pipe(concatMap(r => fileRef.getDownloadURL()));
  }
  getFile(filePath) {
    this.fStorage.ref(filePath).getDownloadURL().subscribe(r => {
      console.log(r)
    })
    // let filePath: string = `${this.uid}/${path}`
    // const fileRef = this.fStorage.ref(filePath);
    // this.downloadURL = fileRef.getDownloadURL();
    // return this.downloadURL.subscribe(r => {
    //   console.log(r)
    // })
  }
}
