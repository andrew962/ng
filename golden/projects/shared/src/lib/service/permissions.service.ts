import { Injectable, OnDestroy } from '@angular/core';
import { AngularFireAuth } from '@angular/fire/auth';
import { AngularFireDatabase } from '@angular/fire/database';
import { Subscription } from 'rxjs';
import { map } from 'rxjs/operators';
import { AuthenticationService } from './authentication.service';

@Injectable({
  providedIn: 'root'
})
export class PermissionsService implements OnDestroy {
  private uid: string;
  authSubscription: Subscription;
  private readonly rule = false
  constructor(
    private auth: AuthenticationService,
    private afDatabase: AngularFireDatabase,
  ) {
    this.authSubscription = this.auth.currentUser$.subscribe(u => this.uid = u?.Uid);
  }
  ngOnDestroy(): void {
    this.authSubscription.unsubscribe();
  }
  setPermissions() {

  }
}
