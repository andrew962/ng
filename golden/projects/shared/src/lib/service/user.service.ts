import { Injectable, OnDestroy } from '@angular/core';
import { AngularFireAuth } from '@angular/fire/auth';
import { AngularFireDatabase } from '@angular/fire/database';
import { Subscription } from 'rxjs';
import { UsersPath } from '../enums/user.enum';
import { CurrentUser } from '../interfaces/currentUser.model';
import { AuthenticationService } from './authentication.service';

@Injectable({
  providedIn: 'root'
})
export class UserService implements OnDestroy {
  private readonly rule = false
  private uid: string;
  user: CurrentUser;
  authSubscription: Subscription;
  constructor(
    private afAuth: AngularFireAuth,
    private db: AngularFireDatabase,
    private auth: AuthenticationService
  ) {
    this.authSubscription = this.auth.currentUser$.subscribe(u => {
      this.user = u;
      this.uid = u?.Uid
    });
  }
  ngOnDestroy(): void {
    this.authSubscription.unsubscribe();
  }

  createUser() {
    this.db.database.ref(UsersPath.DbUsersPath).child(this.uid).set({
      User: this.user,
      Permission: {
        Create: false,
        Update: false,
        Delete: false
      },
      Photos: []
    })
  }
  getUsers() {
    return this.db.database.ref(UsersPath.DbUsersPath).get()
  }
  updateUsers(data) {
    return this.db.database.ref(UsersPath.DbUsersPath).set(data);
  }
  updateUser() {
    this.db.database.ref(UsersPath.DbUsersPath).child(this.uid).update({
      Permission: {
        Create: true,
        Update: true,
        Delete: true
      }
    })
  }

}
