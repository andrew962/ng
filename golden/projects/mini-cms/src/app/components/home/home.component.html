<!-- <div class="col justify-content-center mt-3">
    <mat-form-field class="w-100">
        <mat-label>Buscar</mat-label>
        <input matInput>
      </mat-form-field>
</div> -->